src/pages/Reports/Reports.tsx

```diff
@@ -205,6 +212,53 @@ class Reports extends React.Component<Props, State> {
 
     // Actions
 
+    public handleReportKebabToggle = (report: Report, isOpen: boolean) => {
+        const { toggleReportsIDS } = this.state;
+
+        const newValue = new Set(toggleReportsIDS);
+        if (isOpen) {
+            newValue.add(report.id);
+        }  else {
+            newValue.delete(report.id);
+        }
+        this.setState({
+            toggleReportsIDS: newValue
+        }, () => {
+            this.filtersInRowsAndCells();
+        });
+    };
+
+    public handleDownloadReportPayload = (report: Report) => {
+        const {fetchReportPayloadDownloadLink, addNotification} = this.props;
+
+        // close kebab
+        this.handleReportKebabToggle(report, false);
+        
+
+        fetchReportPayloadDownloadLink(report.id).then((response: any) => {
+            if (response && response.value && response.value.data) {
+                const data = response.value.data;
+                if (data.downloadLink) {
+                    const link = document.createElement('a');
+                    link.href = data.downloadLink;
+                    link.setAttribute('download', data.filename);
+                    document.body.appendChild(link);
+                    link.click();
+                    link.remove();
+                } else {
+                    addNotification({
+                        variant: 'danger',
+                        title: 'Could not download payload file',
+                        description: 'Analysis does not have a payload file',
```

Comment:
Talked with @PhilipCattanach about this and it sounds better changed to be:
```suggestion
                        description: 'The retention period for the payload file has expired',
```